File: solgen.portal/ClientApp/src/app/views/superadmin/super-admin-subscriptions/subscription.service.ts
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';

import { Injectable } from '@angular/core';
import { environment } from '../../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class SubscriptionService {
  public baseUri = `${environment.WebApiBaseUrl}`;
  constructor( private HttpClient:HttpClient) { }
  // GetPackageList(){
  //   return this.HttpClient.get(`${this.baseUri}subscription/GetPackageList`);
  // }
  // GetAdOnsList(){
  //   return this.HttpClient.get(`${this.baseUri}subscription/GetAdOnsList`);
  // }
  GetPackage(){
    return this.HttpClient.get(this.baseUri+"subscription/GetPackage")
  }
  GetAddOn(){
    return this.HttpClient.get(this.baseUri+"subscription/GetAddOn")
  }
  GetDiscount(){
    return this.HttpClient.get(this.baseUri+"subscription/GetDiscount")
  }
  CheckEmailDuplicate(Email: any) {
    return this.HttpClient.get(this.baseUri + `subscription/CheckEmailDuplicate?Email=${Email}`);
  }
  AddSubscription(json:string){
    ;
    return this.HttpClient.post(`${this.baseUri}subscription/AddCompanySubscription`,json);
  }
  GetSubscriptionList(sortCol:string,sortOrder:string,totalPage:number,pageNum:number,searchBy:string,searchIndustry:string, isForDashboard?:boolean,fromDate :string = null,toDate :string = null):Observable<any>{
    return this.HttpClient.get(`${this.baseUri}subscription/GetSubscriptionList?sortCol=${sortCol}&sortOrder=${sortOrder}&totalPage=${totalPage}&pageNum=${pageNum}&searchBy=${searchBy}&searchIndustry=${searchIndustry}&isForSuperAdminDashboard=${isForDashboard}&fromDate=${fromDate}&toDate=${toDate}`);
  }
  DeleteSubscription(Id:number){
    return this.HttpClient.get(this.baseUri + `subscription/DeleteSubscription?Id=${Id}`);
  }
  deleteselectedSubs(ids: string):Observable<any> {
    return this.HttpClient.post(`${this.baseUri}subscription/deleteselectedSubs`, null, {
      params: { ids },
    });
  }
  GetResetPassword(email:string):Observable<any>
  {
   return this.HttpClient.get(this.baseUri+`subscription/GetResetPassword?email=${email}`);
  }

}

File: solgen.Service/Subscription/SubscriptionService.cs
using Solgen.Core;
using Solgen.Core.Models;
using Solgen.Repository.Subscription;
using System;
using System.Collections.Generic;
using System.Text;

namespace Solgen.Service.Subscription
{
    public class SubscriptionService : ISubscriptionService
    {
        private readonly ISubscriptionRepository _subscriptionRepository;
        public SubscriptionService(ISubscriptionRepository subscriptionRepository)
        {
            _subscriptionRepository = subscriptionRepository;
        }

        public int AddCompanySubscription(string UserId, CompanySubscriptionModel model)
        {
           return  _subscriptionRepository.AddCompanySubscription(UserId, model);
        }

        public PagedData GetSubscriptionList(string sortCol, string sortOrder, int totalPage, int pageNum, string searchBy,string searchIndustry, bool? isForSuperAdminDashboard, string fromDate, string toDate)
        {
            return this._subscriptionRepository.GetSubscriptionList(sortCol, sortOrder, totalPage, pageNum, searchBy, searchIndustry, isForSuperAdminDashboard, fromDate, toDate);
        }
        public string CheckEmailDuplicate(string Email)
        {
            return this._subscriptionRepository.CheckEmailDuplicate(Email);
        }

        public bool DeleteSubscription(int Id)
        {
            return this._subscriptionRepository.DeleteSubscription(Id);
        }

        public string CreateCompany(int Id)
        {
            return this._subscriptionRepository.CreateCompany(Id);
        }
        public string GetSubscriptionApi(string companyId)
        {
            return this._subscriptionRepository.GetSubscriptionApi(companyId);
        }
        //public string GetAdOnsList()
        //{
        //    return this._subscriptionRepository.GetAdOnsList();
        //}

        //public string GetPackageList()
        //{
        //    return this._subscriptionRepository.GetPackageList();
        //}
    }
}
